examples/gallery/src/icon-layer.html

```diff
@@ -0,0 +1,191 @@
+  </body>
+
+  <script type="text/javascript">
+
+    const octokit = new Octokit();
+
+    const {DeckGL, IconLayer, LineLayer, ScatterplotLayer, TextLayer, COORDINATE_SYSTEM} = deck;
+    
+    const linkAngles = range(0, 360, 12);
+
+    const deckgl =  new DeckGL({
+      views: [new OrthographicView()],
+      viewState: { target: [0, 0, 0], zoom: 2 },
+      controller: false,
+      layers: [],
+    })
+
+    function range(start, stop, step) {
+      step = step || 1;
+      const array = [];
+      for (let i=start; i<stop; i+=step){
+        array.push(i);
+      }
+      return array;
+    }
+
+    function addTargetPositionToContributor(contributor, linkDistance) {
+      const randomIndex = Math.floor(Math.random() * linkAngles.length);
+      const angle = linkAngles[randomIndex] * Math.PI / 180;
+      const x = Math.cos(angle) * linkDistance;
+      const y = Math.sin(angle) * linkDistance;
+      contributor['targetPosition'] = [ x, y ];
+      linkAngles.splice(randomIndex, 1);
+    }
+
+    function calculateMedian(inputArray) {
+      const array = [...inputArray];
+      if(array.length === 0) {
+        return 0;
+      }
+      array.sort((a, b) => a - b);
+      const medianIndex = Math.floor(array.length / 2);
+      if (array.length % 2) {
+        return array[medianIndex];
+      }
+      return (array[medianIndex - 1] + array[medianIndex]) / 2.0;
+    }
+
+    function calculateQuartiles(array) {
+      const secondQuartile = calculateMedian(array);
+      const firstHalfOfArray = array.filter(number => number < secondQuartile);
+      const secondHalfOfArray = array.filter(number => number > secondQuartile);
+
+      const firstQuartile = calculateMedian(firstHalfOfArray);
+      const thirdQuartile = calculateMedian(secondHalfOfArray);
+
+      return {firstQuartile, secondQuartile, thirdQuartile};
+    }
+
+    function updateTooltip({x, y, object}) {
+      const tooltip = document.getElementById('tooltip');
+      if (object) {
+        tooltip.style.top = `${y}px`;
+        tooltip.style.left = `${x}px`;
+        tooltip.innerHTML = `
+        <div><b>${object.login}</b></div>
+        <div>Contributions: ${object.contributions}</div>`;
+      } else {
+        tooltip.innerHTML = '';
+      }
+    }
+
+    function renderLayers(data) {
+      const sourcePosition = [0, 0];
+      const listOfContributions = data.map(contributor => contributor.contributions);
+      const {firstQuartile, secondQuartile, thirdQuartile} = calculateQuartiles(listOfContributions);
+      const ninetiethPercentile = [...listOfContributions]
+        .sort((a, b) => a - b)[Math.floor(0.90 * listOfContributions.length)];
+
+      listOfContributions.forEach((number, index) => {
+        const linkDistance = 20 * ((number - Math.min(...listOfContributions))
+         / (Math.max(...listOfContributions) - Math.min(...listOfContributions))) + 30;
+
+        addTargetPositionToContributor(data[index], linkDistance);
+      })
```

Comment:
In your code, you calculate the min and max for every contribution, which is not good for performance.

You can use a temporary variable to store this `Math.min(...listOfContributions)` and `Math.max(...listOfContributions)`